internals/daemon/api_notices.go

```diff
@@ -92,12 +128,50 @@ func v1GetNotices(c *Command, r *http.Request, _ *UserState) Response {
 		notices = st.Notices(filter)
 	}
 
-	if notices == nil {
-		notices = []*state.Notice{} // avoid null result
-	}
 	return SyncResponse(notices)
 }
 
+// Get the UID of the request. If the UID is not known, return an error.
+func uidFromRequest(r *http.Request) (uint32, error) {
+	_, uid, _, err := ucrednetGet(r.RemoteAddr)
+	if err != nil {
+		return 0, fmt.Errorf("could not parse request UID")
+	}
+	return uid, nil
+}
+
+// Construct the user IDs filter which will be passed to state.Notices.
+func sanitizeUserIDFilter(queryUserID []string) (*uint32, error) {
+	userIDStrs := strutil.MultiCommaSeparatedList(queryUserID)
+	if len(userIDStrs) > 1 {
+		return nil, fmt.Errorf(`may not include more than one "user-id" parameter`)
+	}
+	userIDInt, err := strconv.ParseInt(userIDStrs[0], 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	if userIDInt < 0 || userIDInt > math.MaxUint32 {
+		return nil, fmt.Errorf("user ID is not a valid uint32: %d", userIDInt)
+	}
+	userID := uint32(userIDInt)
+	return &userID, nil
+}
+
+// Validate the "select" query parameter.
+func sanitizeSelectFilter(querySelect []string) error {
+	selectStrs := strutil.MultiCommaSeparatedList(querySelect)
+	for _, selectStr := range selectStrs {
```

Comment:
I know it's not *quite* as strict, but I'd probably just use `query.Get("select")` in the caller so we're only dealing with one value. That's what the likes of `api_changes.go` do.